FF1Lib/Classes.cs

```diff
@@ -340,39 +340,177 @@ public void SetMpGainOnMaxGain(Flags flags, FF1Rom rom)
 
 		public void Randomize(Flags flags, MT19337 rng, List<string> olditemnames, ItemNames itemnames, FF1Rom rom)
 		{
-			if (!(bool)flags.RandomizeClass)
-			{
+			if (!(bool)flags.RandomizeClass && !(bool)flags.Transmooglifier && !(bool)flags.RandomizeClassChaos)
 				return;
-			}
 
-			// Strings to build info screen in game
+			RandomizeClassHacks(flags, rom);
+
+			if ((bool)flags.Transmooglifier)
+				Transmooglify(flags, rng, rom);
+			else if ((bool)flags.RandomizeClass)

```

Comment:
RandomizeClassChaos can only be activated if RandomizeClass is. As such you'll have to swap the branches here. Currently Chaos classes doesn't work anymore.

Reply:
Sorry, in my local I have everything placed and with mutual exclusions in the classes tab, like this:
image

![](https://user-images.githubusercontent.com/117361/189981540-b32e414d-66fc-4a1c-a4ff-6bb024525481.png)

However I was told to not change tabs and to just put mine in the Beta tab so that's what I did. Can you advise further?

Reply:
just fix the if statement, so that it works with the UI the way it is. somewhat like RandomizeClasses && !ChaosClasses

Reply:
But that's not how the code works? It did not work that way to begin with. Chaos class and Transmooglifier are both mutally exclusive with Blursings, not depending on Blursings being active. The previous UI was simply wrong - if you activated Chaos Mode no blursings would take effect, they were completely different code branches that Chaos Mode hijacked to borrow the dataScreen. I abstracted dataScreen out so I could use it too, and the updated UI correctly reflects their existing relationship in the live build and in the one in the PR.

Reply:
I guess to be clearer:

- If you activated Blursings you got Blursings.
- If you then clicked Chaos Mode, blursings did nothing and Chaos Mode was active
- Likewise, the new feature Transmooglifier also does not activate Blursings

All three are mutally exclusive Class Randomization options, and the new UI reflects this accurately, the previous one from current dev branch was mistaken.